### PR TITLE
perf(deepseek-v4): sanitize SWA slot mapping once per step

### PR DESCRIPTION
**Pull request by Xiangyi1996**
## Summary

The SWA write-slot mapping is computed once per step in `DeepseekV4Model.forward`, but every attention layer re-ran the same sanitize (mask invalid CUDA-graph tokens + capacity clamp, ~7 tiny elementwise kernels) on this layer-invariant mapping. Inside the captured MTP decode graph this baked ~430 redundant kernel nodes that re-executed on every replay.

- Move sanitization into `_deepseek_v4_swa_slot_mapping` so both producers (model-level hoist and the attention-local fallback) return a mapping with invalid tokens and out-of-capacity slots already masked to `-1`; drop the per-layer sanitize from `_insert_swa_cache`.
- Add pool property `swa_capacity_slots` (all layers share one SWA buffer shape). Sanitization fails closed: zero capacity masks every slot, and a pool without the property raises instead of skipping the bounds check.
- Numerics are unchanged: the same mask/clamp runs, once per step instead of once per layer.

## Evidence

Mechanism (exact decode nsys trace, TP8 p4/n8, B200, node-mode; measured on the current V4 integration stack since V4 TP8 serving requires #563/#580/#547):

| | before | after | delta |
| --- | ---: | ---: | ---: |
| decode graph kernel nodes | 4,102 | 3,682 | -420 (matches 61 layers x ~7 kernels) |
| bookkeeping elementwise nodes | 743 | 323 | -420 |
| bookkeeping replay time | 0.842 ms/step | 0.381 ms/step | -0.46 ms/step |
| Memory/elementwise category | 1.982 ms/step | 1.554 ms/step | -21.6% |
| median decode cadence | 20.66 ms | 20.23 ms | -2.1% |

All other kernel categories moved less than +/-0.02 ms/step, so the change is surgical.

Endpoint (adjacent non-profile p4 A/B, same node/container/protocol): neutral within run noise — TPS/GPU +1.00%, TPOT +0.23%, MTP acceptance and cache hit matched (87/87 both arms). The ~2% cadence saving is below single-pair noise resolution, so this PR claims the mechanism win, not an endpoint win.

Tests: `test_deepseek_v4_{config,attention_ops,mtp_prefix_cache}.py` pass on this branch (133 passed); on the integration stack the same change passes the fuller 181-test set. New tests cover mask+capacity sanitization and the fail-closed capacity contract.

## Notes for review

- The remaining 323 baked bookkeeping nodes (~0.38 ms/step) are per-key indexer plan / dense-indices builds captured at graph-capture time; they are a separate, larger change and intentionally out of scope here.
- Graph-replay write-safety test added (producer -> sanitize -> replay -> cache write, sentinel-checked); test_deepseek_v4_attention_ops.py and test_deepseek_v4_mtp_prefix_cache.py are now registered in the runtime-1gpu CUDA CI lane (previously never executed in CI).
- Rebased onto current main. A p4 correctness sanity run (main+this branch, 87/87) is in progress; result will be posted in a comment.

### CODE REVIEW

File: test/runtime/test_deepseek_v4_mtp_prefix_cache.py [hunk 1]
```diff
@@ -15,14 +15,18 @@
...
 import torch
+from ci_system.ci_register import register_cuda_ci
 
 from tokenspeed.runtime.execution.forward_batch_info import ForwardMode
 from tokenspeed.runtime.execution.model_executor import (
     _draft_idle_global_num_tokens_for_step,
 )
 from tokenspeed.runtime.models.deepseek_v4 import _deepseek_v4_swa_slot_mapping
 
+register_cuda_ci(est_time=30, suite="runtime-1gpu")
```
[the remark this answers is not in the document]
Comment (Xiangyi1996) [reply test/runtime/test_deepseek_v4_mtp_prefix_cache.py]: Fixed in b3bb295 — added a `pytest.main` entry point (mirrors test_max_new_tokens_context_cap.py) and switched both new registrations to the repo-standard `sys.path.insert` import pattern; the plain `ci_system` import was also what failed the runtime-1gpu lane on all four runners (ModuleNotFoundError when launched as `python3 <path>`). Verified locally with the exact CI launch mode: this file now runs 6 tests via pytest, attention_ops runs 30 via unittest.
